### PR TITLE
Random address tests

### PR DESCRIPTION
**Pull request by ThomasPoot**

### CODE REVIEW

File: locust-tasks/tasks.py [hunk 1]
```diff
@@ -1,40 +1,98 @@
...
+        print('address:', address_line)
+
+        done = False
+        i = START_POINT
+        counter = 0
+        mistake_count = 0
+        has_mistook = False
+
+        while not done:
+            print('iterating')
```
Comment (MartinHumphrey) [inline locust-tasks/tasks.py]: Would be an improvement to replace all "print" statements with use of the python logging module and appropriate debug, info, warning, error statements and ability to configure log level.

Comment (ThomasPoot) [reply locust-tasks/tasks.py]: This has been implemented: They all use `log.info` on a logger with the `"task.random"` name. It defaults to displaying nothing.

File: locust-tasks/tasks.py [hunk 2]
```diff
@@ -1,40 +1,98 @@
...
+            else:
+                print('  next character')
+                # The user types the next character without mistakes.
+                address_slice = address_line[:i]
+                i += 1
+
+            print('    trying:', address_slice)
+
+            # # Handle results
+            result = requests.get(API_URL + 'addresses/partial' + '?input=' + address_slice + '&limit=' + str(LIMIT) + '&historical=false')
```
[the remark this answers is not in the document]
Comment (ThomasPoot) [reply locust-tasks/tasks.py]: This was a mistake used during debug, and has been fixed. All of the random queries show up as `setup` for the purposes of the test, and can be ignored.

File: locust-tasks/tasks.py [hunk 3]
```diff
@@ -1,40 +1,98 @@
...
-            apiCall = apiPrefix+line.rstrip("\n\r")+"&limit="+LIMIT+"&historical=false"
-            r = self.client.get(apiCall)            
-            assert assertRegEx.search(r.text) is not None, \
-                "Response was invalid as address was not in it"
-            time.sleep(1)
-            #Will exit when list is exhausted then stop
-               
+    def rand_type(self):
+        # # Fetch a target address
+        response = requests.get(API_URL + 'addresses/random')
```
[the remark this answers is not in the document]
Comment (ThomasPoot) [reply locust-tasks/tasks.py]: This was a mistake, and is somewhat vague within the AI codebase. For now, it has been altered to be explicit about historical being false.

File: locust-tasks/tasks.py [hunk 4]
```diff
@@ -1,40 +1,98 @@
...
+import itertools
+import json
+import requests
 from locust import HttpLocust, TaskSet, task
 from locust.exception import StopLocust
 
-TEST_RUN = os.getenv('TEST_RUN', 'postcodes')
-#Set to 100
-LIMIT = os.getenv('LIMIT', "10")
+LIMIT          = os.getenv('LIMIT')           or 10
```
[the remark this answers is not in the document]
Comment (ThomasPoot) [reply locust-tasks/tasks.py]: Corrected.

File: locust-tasks/tasks.py [hunk 5]
```diff
@@ -1,40 +1,98 @@
...
 import re
+import random
 import string
 import time
 import os
+import itertools
+import json
+import requests
 from locust import HttpLocust, TaskSet, task
 from locust.exception import StopLocust
```
[the remark this answers is not in the document]
Comment (ThomasPoot) [reply locust-tasks/tasks.py]: Corrected, apart from `json`, which is used.

File: locust-tasks/tasks.py [hunk 6]
```diff
@@ -1,40 +1,98 @@
 #!/usr/bin/env python3
 import re
+import random
 import string
 import time
 import os
+import itertools
```
[the remark this answers is not in the document]
Comment (ThomasPoot) [reply locust-tasks/tasks.py]: Corrected.

**Pull request merged**